### PR TITLE
[516875] Call prepare for imported projects

### PR DESCRIPTION
**Pull request by rawagner**
Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=516875
Signed-off-by: Rastislav Wagner <rawagner@redhat.com>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=516875

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java [hunk 1]
```diff
@@ -47,35 +44,38 @@ public RestoreProjectJob(IProject project){
 		@Override
 		public IStatus run(IProgressMonitor monitor) {
 			HybridProject hybridProject  = HybridProject.getHybridProject(project);
-			if(hybridProject != null ) {
+			//do not call prepare for project with no engines otherwise cordova will complain
+			
+			if(hybridProject != null){
 				try{
-					hybridProject.prepare(monitor, "");
+					if(hybridProject.getEngineManager().getActiveEngines().length != 0) {
```
Comment (gorkem) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java]: Should this check become a utility on EngineManager?

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/HybridProjectCreator.java [hunk 2]
```diff
@@ -145,6 +145,11 @@ public void convertProject( IProject project , IProgressMonitor monitor) throws
         addPlatformPaths(project, monitor);
         addNature(project, monitor);
         setUpJavaScriptProject(project, monitor);
+        HybridProject newHybridProject = HybridProject.getHybridProject(project);
+        //do not call prepare for project with no engines otherwise cordova will complain
+        if(newHybridProject.getEngineManager().getActiveEngines().length > 0){
```
Comment (gorkem) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/HybridProjectCreator.java]: Same here.. Should it be a utility?

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java [hunk 3]
```diff
@@ -47,35 +44,38 @@ public RestoreProjectJob(IProject project){
...
-					&& delta.getResource().getType() == IResource.PROJECT) {
-				IProject project = delta.getResource().getProject();
-				try {
-					if (project.hasNature(HybridAppNature.NATURE_ID)) {
-						RestoreProjectJob job = new RestoreProjectJob(project);
-						ISchedulingRule rule = ResourcesPlugin.getWorkspace().getRuleFactory()
-								.modifyRule(project);
-						job.setRule(rule);
-						job.schedule();
+			if(delta.getResource().getType() == IResource.PROJECT){
```
Comment (gorkem) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java]: We can possibly collate this with the following if statement?

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java (hunk 1 above)
Comment (rawagner) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java]: added hasActiveEngine() to EngineManager

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java (hunk 3 above)
Comment (rawagner) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/internal/project/RestoreProjectListener.java]: fixed

File: plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/HybridProjectCreator.java (hunk 2 above)
Comment (rawagner) [inline plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/HybridProjectCreator.java]: added hasActiveEngine() to EngineManager

**Comment by rawagner** [comment]
I also added a few more tests

**Pull request merged**